tools/openshift/update-configmap-backend.sh

```diff
@@ -0,0 +1,142 @@
+fi
+
+echo Fetching SOAM token
+TKN=$(curl -s \
+  -d "client_id=admin-cli" \
+  -d "username=$SOAM_KC_LOAD_USER_ADMIN" \
+  -d "password=$SOAM_KC_LOAD_USER_PASS" \
+  -d "grant_type=password" \
+  "https://$SOAM_KC/auth/realms/$SOAM_KC_REALM_ID/protocol/openid-connect/token" | jq -r '.access_token')
+
+echo Fetching public key from SOAM
+fullKey=$(curl -sX GET "https://$SOAM_KC/auth/admin/realms/$SOAM_KC_REALM_ID/keys" \
+  -H "Content-Type: application/json" \
+  -H "Authorization: Bearer $TKN" \
+  | jq -r '.keys | .[] | select(.algorithm == "RS256") | .publicKey')
+
+echo Fetching public key from SOAM
+soamFullPublicKey="-----BEGIN PUBLIC KEY----- $fullKey -----END PUBLIC KEY-----"
+newline=$'\n'
+formattedPublicKey="${soamFullPublicKey:0:26}${newline}${soamFullPublicKey:27:64}${newline}${soamFullPublicKey:91:64}${newline}${soamFullPublicKey:155:64}${newline}${soamFullPublicKey:219:64}${newline}${soamFullPublicKey:283:64}${newline}${soamFullPublicKey:347:64}${newline}${soamFullPublicKey:411:9}${newline}${soamFullPublicKey:420}"
+
+echo Generating private and public keys
+ssh-keygen -b 4096 -t rsa -f tempPenBackendkey -m pem -q -N ""
+UI_PRIVATE_KEY_VAL="$(cat tempPenBackendkey)"
+UI_PUBLIC_KEY_VAL="$(ssh-keygen -f tempPenBackendkey -e -m pem)"
+echo Removing key files
+rm tempPenBackendkey
+rm tempPenBackendkey.pub
+
+
+###########################################################
+#Setup for config-maps
+###########################################################
+#### backend configmap
+echo Creating config map "$APP_NAME"-backend-config-map
+oc create -n "$OPENSHIFT_NAMESPACE" configmap "$APP_NAME"-backend-config-map \
+  --from-literal=NODE_ENV=$nodeEnv \
+  --from-literal=LOG_LEVEL=info \
+  --from-literal=SERVER_FRONTEND="https://$BASE_URL" \
+  --from-literal=SOAM_PUBLIC_KEY="$formattedPublicKey" \
+  --from-literal=SOAM_CLIENT_ID="grad-admin-client" \
+  --from-literal=SOAM_CLIENT_SECRET="$SOAM_CLIENT_SECRET" \
+  --from-literal=SOAM_URL="https://soam-$ENV.apps.silver.devops.gov.bc.ca" \
+  --from-literal=SOAM_DISCOVERY="https://soam-$ENV.apps.silver.devops.gov.bc.ca/auth/realms/master/.well-known/openid-configuration" \
+  --from-literal=IDIR_IDP_HINT=keycloak_bcdevexchange_idir \
+  --from-literal=SITEMINDER_LOGOUT_ENDPOINT="$siteMinderLogoutUrl" \
+  --from-literal=ISSUER=GRAD_ADMIN_APPLICATION \
+  --from-literal=SESSION_MAX_AGE='1800000' \
+  --from-literal=TOKEN_EXPIRES_IN='1800000' \
+  --from-literal=UI_PUBLIC_KEY="$UI_PUBLIC_KEY_VAL" \
+  --from-literal=UI_PRIVATE_KEY="$UI_PRIVATE_KEY_VAL" \
+  --from-literal=GRAD_ROLE_ADMIN=GRAD_SYSTEM_COORDINATOR \
+  --from-literal=GRAD_PROGRAM_AREA_BA=GRAD_PROGRAM_AREA_BA \
+  --from-literal=GRAD_ROLE_INFO_OFFICER=GRAD_INFO_OFFICER \
+  --from-literal=REDIS_HOST=redis \
+  --from-literal=REDIS_PORT=6379 \
+  --from-literal=REDIS_PASSWORD="$REDIS_PASSWORD" \
+  --from-literal=BATCH_API_URL="http://educ-grad-batch-graduation-api.77c02f-$ENV.svc.cluster.local:8080/api/v1" \
```

Comment:
is it ok to have namespace ids in the repo?

Reply:
I posed the question in RC to the platform team awhile back and the response was kind of luke warm about this. I have seen other teams using them in version control also which makes me think it's ok. But good catch anyway, I will change this.